src/policy/semantic.rs

```diff
@@ -569,15 +592,16 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
 
     /// Filter a policy by eliminating absolute timelock constraints
     /// that are not satisfied at the given `n` (`n OP_CHECKLOCKTIMEVERIFY`).
-    pub fn at_lock_time(mut self, n: u32) -> Policy<Pk> {
+    pub fn at_lock_time(mut self, n: LockTime) -> Policy<Pk> {
         self = match self {
             Policy::After(t) => {
-                if !timelock::absolute_timelocks_are_same_unit(t, n) {
+                let t = LockTime::from(t);
+                if !t.is_same_unit(n) {
                     Policy::Unsatisfiable
-                } else if t > n {
+                } else if t.to_consensus_u32() > n.to_consensus_u32() {
```

Comment:
This is okay, but I would have expected an API like 
```rust
let cmp_res = t.is_greater_than(n) ; // Any other name.  

match(cmp_res) {
	Some(true),
	Some(false),
	None,
}
```

Rust miniscript/or any other downstream project should need to know that they should check !t.is_same_unit() before comparing. 

Perhaps, something worth looking at 0.30.0

Reply:
We could add a method `is_satisfied_by_lock` in the same manner as `relative::LockTime`.

Reply:
Which would give us the much nicer:
```rust
            Policy::After(t) => {
                let t = LockTime::from(t);
                } else if !t.is_satisfied_by_lock(n) {
                    Policy::Unsatisfiable
                } else {
                    Policy::After(t.into())
                }
            }
```

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/1258

Reply:
Face palm, we actually provide a patter for this in the rustdocs (double face palm, I even wrote the rustdocs and still forgot to use the correct pattern)

I've updated the PR to be 
```rust
            Policy::After(t) => {
                let t = LockTime::from(t);
                let is_satisfied_by = match (t, n) {
                    (Blocks(t), Blocks(n)) => t <= n,
                    (Seconds(t), Seconds(n)) => t <= n,
                    _ => false
                };
                if !is_satisfied_by {
                    Policy::Unsatisfiable
                } else {
                    Policy::After(t.into())
                }
            }
```